### DISCUSSION TITLE
Where should I put my component folder?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#38664] [Help] [answered]
Should a component folder be under Pages?  Or should I put it at the root ?

**Comment by icyJoseph** [comment, marked as answer]
Everything under pages is reachable as a page. 

So I could arrive to `https://example.com/components/Button` for example - I guess this is unwanted, unless, the page we are developing has a concept of showing components, like car parts, or computer components.

If what we mean with components is `UI` building blocks, like React components, then these should be at the root of the project, as a sibling to `page` directory.

**Reply by icyJoseph** [reply]
Like this:

```
.
├── README.md
├── assets
├── components
├── config
├── hooks
├── next-env.d.ts
├── next.config.js
├── package.json
├── pages
├── public
├── service
├── styles
└── yarn.lock
```

**Reply by icyJoseph** [reply]
Or you could use the `src` folder, https://nextjs.org/docs/advanced-features/src-directory

**Comment by GabenGar** [comment]
`/src/components`
